### DISCUSSION TITLE
Ability to Actually Paste from Clipboard instead of Sending Keys Individually

### DISCUSSION THREAD

**Discussion by ineuw** [autokey/autokey#915] [General]
@ineuw — I moved this lost post from from the wiki 
230923 04:38

I'm having the same issue as this [guy](https://groups.google.com/g/autokey-users/c/2ZxvpRy5Iag/m/ctThDLbKBAAJ?pli=1).

I need the ability to script an actual paste from the clipboard instead of sending each keystroke individually from the clipboard.

You see, in remote RDP sessions, I lose keystrokes when sending keys the normal way, but if I could tell my script to paste what's in the clipboard, instead of sending each keystroke, I wouldn't lose any characters.

Since I cannot find anything in the API that does this, I'm having to fill_clipboard in script and manually press ctrl-v after doing so. I'm just trying to automate away that last step of having to press ctrl-v.

I just tried sending ctrl-v like this `keyboard.send_key("<control>v")` and `keyboard.send_key("<ctrl>v")`.

This only outputs one character where the cursor is resting (at the destination), instead of actually pasting the entire clipboard's contents at once.

**Comment by ineuw** [comment]
I lost the name of the original poster, can someone help to retrieve it?

**Comment by josephj11** [comment]
Yup. That's how `keyboard.send_key()` "works".

See the wiki [article](https://github.com/autokey/autokey/wiki/Key-Combinations#the-keyboardsend_key-method).

Use `keyboard.send_keys("<ctrl>+v")` instead.

It's an easy mistake to make because the two API call function names are almost identical. It would be nice if we could rename one of them to something more distinct.

It will be interesting to see how all of this works with Sam's new interfaces for Wayland.
